% Generated by roxygen2: do not edit by hand
% Please edit documentation in R/reTry.R
\name{reTry}
\alias{reTry}
\title{Re-try the call to a function}
\usage{
reTry(fun, ..., tries = 1, intSleep = 0, errorLogging = flog.error)
}
\arguments{
\item{fun}{(function) a function}

\item{...}{arguments passed to \code{fun}}

\item{tries}{(numeric > 0) number of tries}

\item{intSleep}{(numeric >= 0) interval in seconds between tries}

\item{errorLogging}{(function) a function which is called in case of an error with
\code{errorLogging(<try-error>, ...)}}
}
\description{
Tries to call \code{fun}. If the call results in an error, the error is
logged and after some time it is tried again. If the final try results in an
error the function throws an error. Otherwise th result of fun is returned.
}
\examples{
## Something that will fail every once in a while:
try(reTry(
  function() if (runif(1) > 0.5) stop(),
  tries = 2,
  intSleep = 1
))

## Disable logging
noErrorLogging <- function(x, ...) NULL
try(reTry(
  function() stop(),
  errorLogging = noErrorLogging
))

}

